src/Components/WebAssembly/Server/src/ContentEncodingNegotiator.cs

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Server
+{
+    internal class ContentEncodingNegotiator
+    {
+        // List of encodings by preference order with their associated extension so that we can easily handle "*".
+        private static readonly StringSegment [] _preferredEncodings =
+            new StringSegment [] { "br", "gzip" };
+
+        private static readonly IDictionary<StringSegment, string> _encodingExtensionMap = new Dictionary<StringSegment, string>(StringSegmentComparer.OrdinalIgnoreCase)
+        {
+            ["br"] = ".br",
+            ["gzip"] = ".gz"
+        };
+
+        private readonly RequestDelegate _next;
+
+        public IWebHostEnvironment WebHostEnvironment { get; }
```

Comment:
Can this be a `private readonly` field? Didn't spot it being read from outside.

Reply:
It's on an internal class :). For future reference, I normally `ctrl+.` my way to victory through these internal classes, if you feel strongly about it just let me know and I'll switch to making them fields. Both options are fine and I don't have a strong preference, just want that whatever we choose is consistent across the team.